decentraland/unity-renderer
# refactor: new plugin system

## What does this PR change?

This PR addresses the first item on https://github.com/decentraland/unity-renderer/issues/1269 by integrating a new plugin system implementation.

The new plugin system has the following features:
- Full, easy configurable support for feature flags.
- Single feature flag can share many plugins if needed.
- Simplified plugins lifecycle. When they are enabled, they are instantiated. When they are disabled, they are disposed.
- Improved dependency injection support. The plugins can be initialized with constructor arguments.
- The plugin configuration is provided from outside the `PluginSystem`. Now each supported platform can have their own plugins.

Also `EventUpdateHandler` is introduced to deal with `Update()`, `LateUpdate()` and `OnGUI()` calls, as these are now falling outside `PluginSystem` responsibilities.


## How to test the changes?

A test is to ensure all the feature flags enable the corresponding plugins by firing the build and looking at the taskbar.

Also, the `Update()` changes can affect builder in world. 

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md
